lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -466,28 +466,29 @@ export default class BitcoinProcessor {
       lastValidBlock = await this.revertDatabases();
     }
 
-    // Our starting block is the one after the last-valid-block
-    return this.bitcoinClient.getBlockInfoFromHeight(lastValidBlock.height + 1);
+    // If there is a valid processed block, we will start processing the block following it, else start processing from the genesis block.
+    const startingBlockHeight = lastValidBlock ? lastValidBlock.height + 1 : this.genesisBlockNumber;
+
+    return this.bitcoinClient.getBlockInfoFromHeight(startingBlockHeight);
   }
 
   private async getStartingBlockForPeriodicPoll (): Promise<IBlockInfo | undefined> {
 
-    const lastProcessedBlockVerified = await this.verifyBlock(this.lastProcessedBlock!.height, this.lastProcessedBlock!.hash);
+    const lastProcessedBlockValid = await this.verifyBlock(this.lastProcessedBlock!.height, this.lastProcessedBlock!.hash);
```

Comment:
Generally I'm a bit wary of the revertDatabases() returning undefined. So on line 483, the `lastProcessedBlock` could be undefined but on line 477 here the `lastProcessBlock!` is asserted. I think it will be better if each routine just returns the 'correct' last valid block.

Reply:
One of the main point for this PR is to call out that there is no such "last valid block" if entire DB is reverted, in which case `undefined` to me is the right value to be returned by `revertDatabases()`.

The ! asserts here is not related to the PR change and I tried not to touch that part of the code, this was just a variable rename.

Reply:
Synced up offline and took a valid suggestion of not assigning `this.lastProcessedBlock` to undefined.